ci-operator/step-registry/upi/conf/gcp/upi-conf-gcp-commands.sh

```diff
@@ -60,15 +65,38 @@ data = yaml.load(open(path));
 data["compute"] = [ { "name": "worker", "replicas": 0 } ];
 open(path, "w").write(yaml.dump(data, default_flow_style=False))'
 
+### Update base domain setting
+if [[ -v IS_XPN ]] && [[ -f "${CLUSTER_PROFILE_DIR}/ci-xpn.json" ]]; then
+  echo "Updating base domain (cluster profile 'gcp-qe')..."
+  python -c '
+import yaml;
+import sys;
+path = "install-config.yaml";
+data = yaml.load(open(path));
+data["baseDomain"] = sys.argv[1];
+open(path, "w").write(yaml.dump(data, default_flow_style=False))' "${BASE_DOMAIN}"
+fi
+
 ### Enable private cluster setting (optional)
+origin_publish="External"
 if [[ -v IS_XPN ]]; then
   echo "Enabling private cluster setting..."
   python -c '
 import yaml;
 path = "install-config.yaml";
 data = yaml.load(open(path));
+
+if "publish" in data:
+  f = open("origin_publish", "w")
```

Comment:
I guess we can directly get the value of `publish` setting, then overwrite `origin_publish`, right, instead of writing a file, then load the value from there.